assets/app/features/upgrade/controllers/upgrade-upgrade-administration-server.controller.js

```diff
@@ -48,21 +52,25 @@
                 .then(
                     // In case of success
                     function (response) {
-                        // TODO: update to match API
-                        vm.adminUpgrade.completed = response.data.completed;
+                        // map api response to model
+                        vm.adminUpgrade.completed = response.data.upgrade.success;
+                        vm.adminUpgrade.running = response.data.upgrade.upgrading;
                     },
                     // In case of failure
                     function (errorResponse) {
                         // Expose the error list to adminUpgrade object
+                        // TODO(itxaka): Use the proper error key from the response
                         vm.adminUpgrade.errors = errorResponse.data.errors;
                     }
                 ).finally(
                     function () {
                         // schedule another check if not completed yet
-                        if (!vm.adminUpgrade.completed) {
+                        if (!vm.adminUpgrade.completed && vm.adminUpgrade.running) {
                             $timeout(vm.adminUpgrade.checkAdminUpgrade, 1000); // TODO: make the interval a constant
-                        } else {
-                            vm.adminUpgrade.running = false;
+                        } else if (vm.adminUpgrade.completed && !vm.adminUpgrade.running) {
+                            // TODO(itxaka): can translations be done from here instead so
+                            // we can update the button text for better UX?
+                            vm.finish_text = 'Upgrade done!'
```

Comment:
The translation service can be injected into controllers and used to send text to the view (see https://angular-translate.github.io/docs/#/api/pascalprecht.translate.$translate), but I'd rather use such case only if no other option is possible.
I'd prefer using the VM to expose the state to the view and let it decide what to show, where and when.

Another possibility is to let the controller build and update language keys to be exposed to the view, and use the translations there.